Toggl.Droid/Services/TimerBackgroundService.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using Android.App;
+using Android.Content;
+using Android.Support.V4.App;
+using Toggl.Core;
+using Toggl.Core.Analytics;
+using Toggl.Core.Extensions;
+using Toggl.Core.Interactors;
+
+namespace Toggl.Droid.Services
+{
+    [Service(
+        Exported = true,
+        Name = "com.toggl.giskard.TimerBackgroundService",
+        Permission = "android.permission.BIND_JOB_SERVICE"
+    )]
+    public class TimerBackgroundService : JobIntentService
+    {
+        public const string StartTimeEntryAction = nameof(StartTimeEntryAction);
+        public const string StopRunningTimeEntryAction = nameof(StopRunningTimeEntryAction);
+
+        private IInteractorFactory interactorFactory => AndroidDependencyContainer.Instance.InteractorFactory;
+        private ITimeService timeService => AndroidDependencyContainer.Instance.TimeService;
+
+        public static void EnqueueStartTimeEntry(Context context, Intent intent)
+        {
+            var serviceClass = Java.Lang.Class.FromType(typeof(TimerBackgroundService));
+            EnqueueWork(context, serviceClass, JobServicesConstants.TimerWidgetStartTimeEntryJobId, intent);
+        }
+
+        public static void EnqueueStopTimeEntry(Context context, Intent intent)
+        {
+            var serviceClass = Java.Lang.Class.FromType(typeof(TimerBackgroundService));
+            EnqueueWork(context, serviceClass, JobServicesConstants.TimerWidgetStopRunningTimeEntryJobId, intent);
+        }
+
+        protected override void OnHandleWork(Intent intent)
+        {
+            var action = intent.Action;
+
+            AndroidDependencyContainer.EnsureInitialized(Application.Context);
+
+            switch (action)
+            {
+                case StartTimeEntryAction:
+                    handleStartTimeEntry();
+                    break;
+                case StopRunningTimeEntryAction:
+                    handleStopRunningTimeEntry();
```

Comment:
Discard this result, to avoid the async in non-async method warnings and to signify that you are aware that the task is running async.
```suggestion
                    _ = handleStopRunningTimeEntry();
```